src/detect-engine-port.c

```diff
@@ -1290,6 +1290,11 @@ DetectPort *PortParse(const char *str)
     if (dp == NULL)
         goto error;
 
+    // Bug fix for strlen(str) >= 16
```

Comment:
if we allow leading spaces, and we probably should, shouldn't we do something like
```c
     while (isspace(*str))
         str++;
     if (strlen(str) >= 16)
         return NULL;
```